mechafil/power.py

```diff
@@ -1,17 +1,30 @@
 import numpy as np
 import pandas as pd
 import datetime
-from typing import Callable, Tuple
+from typing import Callable, Tuple, Union
+import numbers
+
+# --------------------------------------------------------------------------------------
+#  Utility functions
+# --------------------------------------------------------------------------------------
+def scalar_or_vector_to_vector(input_x, expected_len, err_msg=None):
+    if isinstance(input_x, numbers.Number):
+        return np.ones(expected_len) * input_x
+    else:
+        err_msg_out = "vector input does not match expected length!" if err_msg is None else err_msg
```

Comment:
Why not make "vector input does not match expected length!" the default input of `err_msg`?

Reply:
Mainly for cosmetic reasons, since that would be a really long string in the function header? But, open to changing if you think otherwise. Strings are immutable so I don't think there is a concern as there would be if passing in a dictionary.